ios-swiftui/realm-template-apps-swiftui/Views/LogoutButton.swift

```diff
@@ -4,19 +4,54 @@ import RealmSwift
 /// Logout from the synchronized realm. Returns the user to the login/sign up screen.
 struct LogoutButton: View {
     @State var isLoggingOut = false
+    @State var error: Error?
+    @State var errorMessage: ErrorMessage? = nil
     
     var body: some View {
+        if isLoggingOut {
+            ProgressView()
+        }
         Button("Log Out") {
             guard let user = realmApp.currentUser else {
```

Comment:
This is probably why the button doesn't work after the first logout. Are you sure logOut() does not clear the local realmApp.currentUser state regardless of whether it actually succeeding in telling the server to log out? (I'm not even sure how much it needs to do on the server side anyway, since "logging out" could just be deleting your access tokens)

Reply:
Ahh, you're right - `logOut()` does clear the local authentication state even if there is an error. So is it meaningful to display an error here at all, since the user's auth state is cleared regardless of whether `logOut()` returns an error?

Reply:
It might be worth displaying an error but logout is successful if the current user is cleared. So this should cause the app to go back to the login screen.

Reply:
I think I'm missing something. Even when `logOut()` errors, it's clearing the app user. But is not going back to the LoginView(). I've checked for `app.currentUser` after the error, and there isn't one. I think I am stumped. Any suggestions?

Reply:
I believe we still log out unless it's a network error. It's possible in the event of an error, the correct is not being propagated however.